packages/ketcher-react/src/script/editor/tool/select.js

```diff
@@ -264,6 +270,15 @@ SelectTool.prototype.dblclick = function (event) {
   } else if (ci.map === 'sgroups' || ci.map === 'sgroupData') {
     this.editor.selection(closestToSel(ci))
     sgroupDialog(this.editor, ci.id)
+  } else if (ci.map === 'texts') {
+    this.editor.selection(closestToSel(ci))
+    const text = struct.texts.get(ci.id)
+    const dialog = editor.event.elementEdit.dispatch(text)
+    Promise.resolve(dialog)
+      .then(newText => {
+        editor.update(fromTextUpdating(rnd.ctab, ci.id, newText))
+      })
+      .catch(() => null)
```

Comment:
Why do you need `Promise.resolve(dialog)` ?

Reply:
Thank you for the explanation during our call, refactored